pkg/controller/chaosengine/chaosengine_controller.go

```diff
@@ -228,29 +236,53 @@ func (r *ReconcileChaosEngine) Reconcile(request reconcile.Request) (reconcile.R
 	}
 	// Define the engine-monitor service which is secondary-resource #2
 	engineMonitor, err := newMonitorServiceForCR(instance)
+	if !instance.Spec.MonitorEnable {
+		reqLogger.Info("No EngineMonitor SVC Because EngineMonitor DISABLED")
+	}
 	if err != nil {
 		return reconcile.Result{}, nil
 	}
+	// Define an engine-exporter pod which is secondary-resource #3
+	engineExporter, err := newExporterPodForCR(instance, appUUID)
+	if !instance.Spec.MonitorEnable {
+		reqLogger.Info("No EngineExporter POD Because EngineMonitor DISABLED")
+	}
+	if err != nil {
+		return reconcile.Result{}, nil
+	}
+	if instance.Spec.MonitorEnable {
```

Comment:
Would it make more sense to create the engineExporter & set controller reference all in one block when instance.Spec.MonitorEnable is true? --> rather than do a negative check, print a log, create resource and then set controller reference if enabled?

Reply:
I tried it, but return statements are causing issues